plone/app/contenttypes/browser/collection.py

```diff
@@ -13,9 +13,17 @@ class CollectionView(FolderView):
     def __init__(self, *args, **kwargs):
         super(CollectionView, self).__init__(*args, **kwargs)
         context = aq_inner(self.context)
-        self.collection_behavior = ICollection(context)
+
+        try:
+            # the object may have the behavior applied
+            self.collection_behavior = ICollection(context)
```

Comment:
Adaption must always work. If not there is something different wrong. This looks like a workaround for a different unsolved problem.